File: components/core/src/main/java/org/storm/core/ui/Resolutions.java
package org.storm.core.ui;

import java.awt.*;

/**
 * A collection of common resolutions
 */
public class Resolutions {

  // Common Resolutions
  public static final Resolution SD = new Resolution(640, 480);
  public static final Resolution HD = new Resolution(1280, 720);
  public static final Resolution FHD = new Resolution(1920, 1080);
  public static final Resolution QHD = new Resolution(2560, 1440);
  public static final Resolution UHD = new Resolution(3840, 2160);
  public static final Resolution FULL_SCREEN = new Resolution(
    Toolkit.getDefaultToolkit().getScreenSize().getWidth(),
    Toolkit.getDefaultToolkit().getScreenSize().getHeight()
  );


  private Resolutions() {}

}

File: components/core/src/main/kotlin/org/storm/core/ui/Resolutions.kt
package org.storm.core.ui

import java.awt.Toolkit

/**
 * A collection of common resolutions
 * TODO: Remove this class and make everyone use the companion object in Resolution
 */
object Resolutions {
  // Common Resolutions
  val SD = Resolution(640.0, 480.0)
  val HD = Resolution(1280.0, 720.0)
  val FHD = Resolution(1920.0, 1080.0)
  val QHD = Resolution(2560.0, 1440.0)
  val UHD = Resolution(3840.0, 2160.0)
  val FULL_SCREEN = Resolution(
    Toolkit.getDefaultToolkit().screenSize.getWidth(),
    Toolkit.getDefaultToolkit().screenSize.getHeight()
  )
}
